src/java.compiler/share/classes/javax/lang/model/package-info.java

```diff
@@ -46,12 +46,96 @@
+ * shared by all the uses.
+
+ * <p>For example, consider the uses of {@code
+ * java.lang.String} in the string processing method {@code
+ * identityOrEmpty} below:
+ *
+ * {@snippet lang=java :
+ * // Return the argument if it is non-null and the empty string otherwise.
+ * public static @DefinitelyNotNull String identityOrEmpty(@MightBeNull String argument) {
+ *    ...
+ * }
+ * }
+ *
+ * The return type of the method is a {@code String} annotated with
+ * a {@code @DefinitelyNotNull} type annotation while the type of
+ * the parameter is a {@code String} annotated with a {@code
+ * @MightBeNull} type annotation. In a reflective API, since the set
+ * of annotations is different for the two <em>uses</em> of {@code
+ * String} as a type, the return type and argument type would need to
+ * be represented by different objects to distinguish between these two
+ * cases. The <em>definition</em> of {@code java.lang.String} itself
+ * is annotated with neither of the type annotations in question.
+ *
+ * <p>Another example, consider the declaration of the generic
+ * interface (JLS {@jls 9.1.2}) {@code java.util.Set} which has one
+ * type parameter. This declaration captures commonality between the
+ * many parameterized types (JLS {@jls 4.5}) derived from that
+ * declaration such as {@code java.util.Set<String>}, {@code
+ * java.util.Set<E>}, {@code java.util.Set<?>}, and also the raw type
+ * (JLS {@jls 4.8}) {@code java.util.Set}.
+ *
+ * <h3><a id=elementTypeMapping>Mapping between Elements and Types</a></h3>
+ *
+ * While distinct concepts, there are bidirectional (partial) mappings
+ * between elements and types, between definitions and uses. For
+ * example, roughly speaking, information that would be invariant for
+ * all uses of a type can be retrieved from the element defining a
+ * type. For example, consider a {@link
+ * javax.lang.model.type.DeclaredType DeclaredType} type mirror
+ * modeling a use of {@code java.lang.String}. Calling {@link
+ * javax.lang.model.type.DeclaredType#asElement()} would return the
+ * {@link javax.lang.model.element.TypeElement} for {@code
+ * java.lang.String}. From the {@code TypeElement}, common information
+ * such as {@linkplain
+ * javax.lang.model.element.TypeElement#getSimpleName() name} and
+ * {@linkplain javax.lang.model.element.TypeElement#getModifiers()
+ * modifiers} can be retrieved.
+ *
+ * <p>All elements can be {@linkplain
+ * javax.lang.model.element.Element#asType() mapped to} some type.
+ * The elements for classes and interfaces get {@linkplain
+ * javax.lang.model.element.TypeElement#asType() mapped to} a
+ * prototypical type.
+ * (If a class or interface is generic, its prototypical type mirror
+ * is parameterized with type arguments matching the type variables of
+ * the declaration. Otherwise, for a non-generic class or interface,
+ * the prototypical type mirror corresponds to an unannotated use of
+ * the type.)
```

Comment:
Perhaps I'm not quite following it, but why juxtapose "default" parameterization with being unannotated? I assume, for a generic class or interface, the prototypical type is also unannotated, no?

Reply:
I assume Maurizio suggested mentioning annotations given that they were discussed as one reason uses need separate objects.

Adding a note that the other case is unannotated as well.